File: backend/product-service/src/main/java/com/mall/product/entity/Category.java
package com.mall.product.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("category")
public class Category {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private Long parentId;
    private Integer sort;
    @TableField("icon")
    private String imageUrl;
    @TableField(exist = false)
    private Integer status;
    private LocalDateTime createdTime;
    private LocalDateTime updatedTime;

    @TableField(exist = false)
    private List<Category> children;
}

File: admin-frontend/src/api/category.ts
import request from './request'

export interface Category {
  id: number
  name: string
  parentId: number | null
  sort: number
  imageUrl: string | null
  status: number | null
  children?: Category[]
}

export function getCategoryList() {
  return request.get<any, Category[]>('/product/category/admin/list')
}

export function addCategory(data: Partial<Category>) {
  return request.post<void>('/product/category/admin/add', data)
}

export function updateCategory(data: Partial<Category>) {
  return request.put<void>('/product/category/admin/update', data)
}

export function deleteCategory(id: number) {
  return request.delete<void>('/product/category/admin/delete/' + id)
}
